src/main/kotlin/build/buf/intellij/configurable/BufConfigurable.kt

```diff
@@ -67,4 +83,18 @@ class BufConfigurable(
         if (super.isModified()) return true
         return project.bufSettings.state != state
     }
+
+    private fun validateBufCLI(input: TextFieldWithBrowseButton): ValidationInfo? {
+        val value = input.text
+        if (value.isEmpty()) {
+            return null
+        }
+        val file = Paths.get(value).toFile()
+        return when {
+            !file.exists() -> ValidationInfo(BufBundle.message("settings.buf.cli.path.error.not_exist"), input)
+            !file.isFile -> ValidationInfo(BufBundle.message("settings.buf.cli.path.error.not_file"), input)
+            !file.canExecute() -> ValidationInfo(BufBundle.message("settings.buf.cli.path.error.not_executable"), input)
+            else -> null
+        }
```

Comment:
I know I could get more sophisticated here, but this will avoid the majority of issues.